packages/filesystem/src/browser/file-tree/file-tree.ts

```diff
@@ -28,6 +28,30 @@ export class FileTree extends TreeImpl {
     @inject(FileSystem) protected readonly fileSystem: FileSystem;
     @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
 
+    @postConstruct()
+    protected init(): void {
+        this.toDispose.push(
+            this.labelProvider.onElementUpdated((uri: URI) => this.doUpdateElement(uri))
+        );
+    }
+
+    protected doUpdateElement(uri: URI): void {
+        const nodePath = uri.path.toString();
+        if (this.root && CompositeTreeNode.is(this.root)) {
+            for (const workspaceRoot of this.root.children) {
+                if (CompositeTreeNode.is(workspaceRoot)
+                    && nodePath.startsWith(workspaceRoot.id + '/')) {
+                    const id = this.toNodeId(uri, workspaceRoot);
+                    const node = this.getNode(id);
+                    if (node) {
+                        delete this.nodes[id];
```

Comment:
I don't think it is good to remove a node, without properly update other node siblings, parents and so on.

Could we explicitly update `name` and `icon` of a node here instead? It would be good to extract template methods to create `name` and `icon`. And call them from here as well as from `toNode`.